src/index.jsx

```diff
@@ -17,21 +17,22 @@ import configureStore from './store';
 import './index.scss';
 import App from './App';
 
+const rootNode = createRoot(document.getElementById('root'));
 subscribe(APP_READY, () => {
   const { store } = configureStore(process.env.NODE_ENV);
-
-  ReactDOM.render(
-    <AppProvider store={store}>
-      <App />
-    </AppProvider>,
-    document.getElementById('root'),
+  rootNode.render(
+    <StrictMode>
```

Comment:
This looks good overall. I'm curious about the `StrictMode` component. Could you help me understand what it is and why it's necessary for the upgrade?

Reply:
Thanks!